src/addrman.h

```diff
@@ -723,22 +729,19 @@ class CAddrMan
     CAddrInfo SelectTriedCollision_() EXCLUSIVE_LOCKS_REQUIRED(cs);
 
     //! Consistency check
-    void Check()
-        EXCLUSIVE_LOCKS_REQUIRED(cs)
+    void Check() EXCLUSIVE_LOCKS_REQUIRED(cs)
     {
-#ifdef DEBUG_ADDRMAN
         AssertLockHeld(cs);
-        const int err = Check_();
+
+        int err = Check_();
         if (err) {
             LogPrintf("ADDRMAN CONSISTENCY CHECK FAILED!!! err=%i\n", err);
+            assert(false);
```

Comment:
Nice. Verbosely complaining to developers about bugs makes them less likely to get missed.

Reply:
Unrelated improvement: Maybe just simplify the `if foo return -xx;` with `assert(!foo);` in the cpp file?

Reply:
```suggestion
            std::abort();
```

Reply:
@MarcoFalke I left the cpp file unchanged in case we want to later add some kind of recovery code instead of simply asserting on failure. Changing the `if foo return -xx;` to `assert(!foo)` would involve making changes in more places.

@vasild - What's the advantage of using `std::abort`? I see lots more examples of `assert(false)` in the code than `std::abort()`.

Reply:
`assert()` calls `abort()`, so it is the same thing in practice (modulo the "strange" message `assertion failed: false`). Feel free to ignore or use `assert(0)` or `assert(nullptr)`.

Reply:
`assert(false && "addrman consistency check failed");` perhaps? Or `assert(err == 0)` or `Assume(err == 0)`

Reply:
#22500 ;)

Reply:
Thanks @MarcoFalke. I've added the two commits from that PR into this branch.